test/evergreen/tsan_warnings_analysis.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import os
+import re, subprocess
+test_tasks = [
+    "python3 ../test/suite/run.py -j 8 --ignore-stdout",
+    "ctest -j 8 -LE \"long_running\"",
+    "bash test/format/format.sh -j 8 -t 60 rows=10000 ops=50000"
+  ]
+
+# Run the tasks in list.
+task_list = list()
+print(test_tasks)
+for task in test_tasks:
+    try:
+        split_command = task.split()
+        # Tasks are allowed to run with maximum of three hours.
+        subprocess.run(split_command, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, timeout=10800)
+    except subprocess.CalledProcessError as exception:
+        print(f'Command {exception.cmd} failed with error {exception.returncode}')
+    except subprocess.TimeoutExpired as exception:
+        print(f'Command {exception.cmd} timed out')
+
+# Loop through all subdirectories recursively and search tsan logs.
+tsan_warnings_set = set()
+for tsan_log in os.listdir(".."):
+    # Check if the file starts with "tsan."
+    if tsan_log.startswith("tsan"):
```

Comment:
Shouldn't there be only one `tsan_logs` file?

Reply:
The tsan logs are attached with a PID and created per process.